src/main/java/bio/terra/service/auth/iam/IamService.java

```diff
@@ -240,12 +245,27 @@ public Map<IamRole, String> createDatasetResource(
    *
    * @param userReq authenticated user
    * @param snapshotId id of the snapshot
-   * @param readersList list of emails of users to add as readers of the snapshot
-   * @return Policy group map
+   * @param policies user emails to add as snapshot policy members
+   * @return Map of policy group emails for the snapshot policies
    */
   public Map<IamRole, String> createSnapshotResource(
-      AuthenticatedUserRequest userReq, UUID snapshotId, List<String> readersList) {
-    return callProvider(() -> iamProvider.createSnapshotResource(userReq, snapshotId, readersList));
+      AuthenticatedUserRequest userReq, UUID snapshotId, SnapshotRequestModelPolicies policies) {
+    return callProvider(() -> iamProvider.createSnapshotResource(userReq, snapshotId, policies));
+  }
+
+  /**
+   * @param request snapshot creation request
+   * @return user-defined snapshot policy object, supplemented with readers from deprecated input
+   */
+  public SnapshotRequestModelPolicies deriveSnapshotPolicies(SnapshotRequestModel request) {
+    SnapshotRequestModelPolicies policies =
+        Optional.ofNullable(request.getPolicies()).orElseGet(SnapshotRequestModelPolicies::new);
+
+    List<String> combinedReaders = new ArrayList<>();
+    combinedReaders.addAll(ListUtils.emptyIfNull(policies.getReaders()));
+    combinedReaders.addAll(ListUtils.emptyIfNull(request.getReaders()));
```

Comment:
Might be worth just double checking that this works ok if we have duplicates in here (e.g. maybe we need this to be a set?)

Reply:
Can do!  I *think* this is handled appropriately and is checked in automated tests, but even so, that could be made more clear.

Reply:
I verified our handling of duplicates in this integration test:

https://github.com/DataBiosphere/jade-data-repo/pull/1356/files#r986230014

I can expand our unit tests of this method to explicitly call out that deduplication of inputs is not performed here. I can also add a comment to this effect in the method.  I don't want to introduce unnecessary deduplication because this would be a departure in how we pass information to SAM, which I think could confuse matters.  How's that sound?

Reply:
That's perfect.  My main thought was just confirming the behavior.  Thanks so much for looking into it.  I'm cool with you merging

Reply:
Awesome!  I added a comment, but left the unit tests as is :)